src/oauth2/client.cr

```diff
@@ -54,18 +54,27 @@
 # You can also use an `OAuth2::Session` to automatically refresh expired
 # tokens before each request.
 class OAuth2::Client
+  private getter host, client_id, client_secret, port, scheme, authorize_uri,
+    redirect_uri, auth_scheme, token_uri
+
```

Comment:
What's the point of this `private getter` instead of using instance variables?
Smells like YAGNI :)

Reply:
I always prefer to use getters/setters instead of accessing instance variables directly.

Reply:
Well, refactoring like this might be considered out of scope of this PR. Secondly it's more costly, so if there's no good reason I'd advice against that.

Reply:
My reason for preferring getters/setters over instance variables even privately is that if the feature needs to change from being a variable to a method then you restrict your changes to the getter/setter.

Context: I once "won" an argument over this on a project where my view was that we *should* access private instance variables directly, and then lived to regret it.  Since then, I've always had a horror of accessing instance variables directly (outside of constructors/getters/setters).

Reply:
OTOH if you have a horror of private accessors I can change them back.  Your call.

Reply:
I'd suggest doing so, yes :)

Reply:
@Sija fwiw https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/codegen/call.cr#L430-L437

Reply:
@jhass I'd argue that in this case it's simply out of scope of this PR, but thanks for sharing this anyway ❤️

Reply:
@jhass Good to know

Reply:
@Sija private accessors dropped for using instance variables directly.  I won't tell Bertrand Meyer if you don't.